pkg/packagekit/updates.jsx

```diff
@@ -962,13 +1304,41 @@ class OsUpdates extends React.Component {
                                  onCancel={ () => PK.call(this.state.applyTransaction, PK.transactionInterface, "Cancel", []) }
                                  allowCancel={this.state.allowCancel} />;
 
-        case "updateSuccess":
+        case "updateSuccess": {
             page_status.set_own({
                 type: "warning",
-                title: _("Restart recommended")
+                title: this.state.tracerPackages.reboot.length > 0
+                    ? cockpit.ngettext("A package needs a system reboot for the updates to take effect:",
+                                       "Some packages need a system reboot for the updates to take effect:",
+                                       this.state.tracerPackages.reboot.length)
+                    : cockpit.ngettext("A service needs to be restarted for the updates to take effect:",
+                                       "Some services need to be restarted for the updates to take effect:",
+                                       this.state.tracerPackages.daemons.length)
```

Comment:
I wonder where the non-tracer case is handled now.

Reply:
shame, such a stupid error i did there :D